sdr/concepts_and_interactions.md

```diff
@@ -0,0 +1,202 @@
+
+This architecture was chosen before open source workflow automation tools like Airflow were available/mature.
+
+Another thing that adds indirection/confusion: sometimes the robots will perform the heavy lifting of a task, but sometimes even they call out to other services.  For example, in the `preservationIngestWF`, the `validate-moab` step in the preservation_robots actually makes a REST call to preservation_catalog to do the validation asynchronously (because that service will do all auditing and cloud replication after the robot finishes updating the content).  And then preservation_catalog tells workflow service when it's done and workflow service tells the preservation_robots to do the next step.
+
+(ask anyone and hand waving will occur.  Perhaps Peter? Andrew? Naomi? JCoyne? - we can re-figure out the layers of legacy code as a team)
+
+
+## Access Rights for digital content
+
+(ask JCoyne, John)
+
+* What's an APO?
+* Where are access restrictions computed when interacting with a specific object? - various codebases use the logic provided by the dor-rights-auth gem:  https://github.com/sul-dlss/dor-rights-auth
+* How are they stored? - Currently, in the `rightsMetadata` datastream for an object (an XML datstream stored in Fedora for a druid).  Will be migrated to the Cocina data model and its JSON serialization format.
+* How do they work to restrict access?
+  * Defaults are applied based on the item's parent APO
+  * An item can have multiple files, and those files may each have different rights specified from each other as well as from the parent item.  If they don't have rights specified individually, they inherit from the parent item.
+* See also
+  * [Rights metadata and public access to SDR objects](https://consul.stanford.edu/display/SDRdocs/Rights+metadata+and+public+access+to+SDR+objects)
+  * [Rights metadata -- the rightsMetadata datastream](https://consul.stanford.edu/display/chimera/Rights+metadata+--+the+rightsMetadata+datastream)
+  * [Argo - Editing the license and rights statements for a set of objects](https://consul.stanford.edu/display/DLSSDOCS/Argo+-+Editing+the+license+and+rights+statements+for+a+set+of+objects)
+
+## Embargoes
+
+(ask JCoyne)
+
+how are they created, stored, and what processing happens when they expire
+* See integration test for ETD
+* See integration test for H2
+* Expiring - see … cron job for dor-services-app (?)
+
+## SDR APIs
+* dor-services-app https://sul-dlss.github.io/dor-services-app/ (https://github.com/sul-dlss/dor-services-app)
+  * dor-services-client
+* cocina-models: https://sul-dlss.github.io/cocina-models/ (https://github.com/sul-dlss/cocina-models)
+* sdr-api: https://sul-dlss.github.io/sdr-api/ (https://github.com/sul-dlss/sdr-api)
+  * sdr-client
+* preservation-catalog api: https://sul-dlss.github.io/preservation_catalog/ (https://github.com/sul-dlss/preservation_catalog)
+  * preservation-client
+
+## Google-books
```

Comment:
Should these (GB, ETD, preassembly, preservation) be on the service catalog rather than in the concepts?

Reply:
i debated that too, and could definitely move, but my thinking here was that there's a terse pointer to the codebase in the service catalog, and these sections are more about how these services fit into the larger system and what each one's overall goals are (or at least that was the intent; there may be a better way to get that across, or that may not be the right approach).  maybe splitting hairs in a confusing way? 🤷